File: Model/EventResponse.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Auctus.Model
{
    public class EventResponse
    {
        public int EventId { get; set; }
        public DateTime EventDate { get; set; }
        public DateTime CreationDate { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public bool CanOccurBefore { get; set; }
        public string Source { get; set; }
        public List<EventCategoryResponse> Categories { get; set; } = new List<EventCategoryResponse>();

        public class EventCategoryResponse
        {
            public int Id { get; set; }
            public string Name { get; set; }
        }
    }
}

File: Web/src/app/model/asset/eventResponse.ts
export class EventResponse {
  eventId: number;
  title: string;
  description: string;
  eventDate: Date;
  creationDate: Date;
  canOccurBefore: boolean;
  source: string;
  categories: EventCategory[];
}

export class EventCategory
{
    id: number;
    name: string;
}